model/host/host.go

```diff
@@ -1600,6 +1600,37 @@ func (h *Host) AddSSHKeyName(name string) error {
 	return nil
 }
 
+// HandleTerminatedHostSpawnedByTask should be called when a host spawned by a task
+// is found terminated.
+func (h *Host) HandleTerminatedHostSpawnedByTask() error {
+	if !h.SpawnOptions.SpawnedByTask {
+		return nil
+	}
+
+	t, err := task.FindOneIdAndExecution(h.SpawnOptions.TaskID, h.SpawnOptions.TaskExecutionNumber)
+	if err != nil {
+		return errors.Wrapf(err, "finding task for host '%s'", h.Id)
+	}
+	if t == nil {
+		return errors.Errorf("host '%s' created by task '%s' execution '%d' that does not exist", h.Id, h.SpawnOptions.TaskID, h.SpawnOptions.TaskExecutionNumber)
+	}
+
+	if h.Status == evergreen.HostStarting && t.Status == evergreen.TaskStarted && !t.Aborted {
+		intent := newIntentFromHost(h)
+		grip.Info(message.Fields{
+			"message":              "inserting a host intent to replace a terminated host for a task",
+			"original_host":        h.Id,
```

Comment:
nit: Most of the code base use `"host_id"` to reference fields containing host ID fields in grip logs (none of the other DB models include `_id`, but hosts do). Can we also make it consistent here, so it's `"original_host_id"` and `"new_host_id"`?

Reply:
done